get_uru_client

```diff
@@ -0,0 +1,284 @@
+        --force-32bit | --force32bit | -force32bit | -32 )
+            FORCE_32BIT=true
+            shift
+            ;;
+        --destiny )
+            DESTINY=true
+            shift
+            ;;
+        --verbose | -v )
+            set -x
+            shift
+            ;;
+        --help | -help | -h | -? )
+            echo "$USAGE"
+            exit 0
+            ;;
+        -- )
+            shift
+            break
+            ;;
+        * )
+            break
+            ;;
+    esac
+done
+
+DOWNLOAD_DIR="$INSTALL_DIR/downloads"
+ASSETS_DIR="$INSTALL_DIR/downloads/moul-assets"
+CLIENT_DIR="$INSTALL_DIR/client"
+
+ASSETS_REV_FILE="$DOWNLOAD_DIR/moul-assets.rev"
+
+# We don't have a working Linux client, so on non-macOS systems just install
+# the Windows client to run through wine.
+if [ "$(echo $OSTYPE | cut -c 1-6)" != "darwin" ]; then
+    FORCE_WINDOWS=true
+fi
+
+if [ "$FORCE_WINDOWS" = true ]; then
+    if [ "$FORCE_32BIT" = true ] || [ "$MACHINE_ARCH" != "x86_64" ]; then
+        CLIENT_DOWNLOAD_FILENAME="plasma-windows-x86-internal-release.zip"
+    else
+        CLIENT_DOWNLOAD_FILENAME="plasma-windows-x64-internal-release.zip"
+    fi
+elif [ "$(echo $OSTYPE | cut -c 1-6)" = "darwin" ]; then
+    if [ "$MACHINE_ARCH" = "arm64" ]; then
+        CLIENT_DOWNLOAD_FILENAME="plasma-macos-arm64-internal-release.zip"
+    else
+        CLIENT_DOWNLOAD_FILENAME="plasma-macos-x64-internal-release.zip"
+    fi
+else
+    echo "$(tput red)Unsupported system $OSTYPE$(tput off)"
+    exit 2
+fi
+
+
+echo "Running GetUru on $OSTYPE ($MACHINE_ARCH)"
+echo ""
+
+# Check for required software (cURL, Git, Git LFS)
+echo "$(tput cyan)Checking for cURL...$(tput off)"
+if [ -f "$(command -v curl)" ]; then
+    echo "    $(tput green)... found$(tput off)"
+else
+    echo "    $(tput red)... not found. cURL is required$(tput off)"
+    exit 1
+fi
+
+echo "$(tput cyan)Checking for Git...$(tput off)"
+if [ -f "$(command -v git)" ]; then
+    echo "    $(tput green)... found$(tput off)"
+else
+    echo "    $(tput red)... not found. Git is required$(tput off)"
+    exit 1
+fi
+
+echo "$(tput cyan)Checking for Git LFS...$(tput off)"
+if [ -f "$(command -v git-lfs)" ]; then
+    echo "    $(tput green)... found$(tput off)"
+else
+    echo "    $(tput red)... not found. Git LFS is required$(tput off)"
+    exit 1
+fi
+
+
+# Prepare output directories before we get started.
+if [ ! -d "$INSTALL_DIR" ]; then
+    echo "$(tput red)The specified install directory '$INSTALL_DIR' does not exist or is not valid!$(tput off)"
+    exit 3
+fi
+
+if [ ! -d "$DOWNLOAD_DIR" ]; then
+    mkdir -p "$DOWNLOAD_DIR"
+fi
+
+
+# The goal here is to *install* Uru. While we will be using git to facilitate acquiring
+# files, we don't actually care about maintaining a copy of the repos themselves. So,
+# if they already exist, nuke them. An potential improvement would be to let the
+# empty sparse checkout remain behind and check for updates as needed.
```

Comment:
It would be _really nice_ to implement this so we don't have to re-clone moul-assets every time the script runs, but that can probably be handled as a follow-up PR